avogadro/qtplugins/atompairbonds/atompairbonds.cpp

```diff
@@ -0,0 +1,159 @@
+/******************************************************************************
+  This source file is part of the Avogadro project.
+  This source code is released under the 3-Clause BSD License, (see "LICENSE").
+******************************************************************************/
+
+#include "atompairbonds.h"
+
+#include <avogadro/core/array.h>
+#include <avogadro/core/bond.h>
+#include <avogadro/core/elements.h>
+#include <avogadro/core/neighborperceiver.h>
+#include <avogadro/qtgui/molecule.h>
+#include <avogadro/rendering/dashedlinegeometry.h>
+#include <avogadro/rendering/geometrynode.h>
+#include <avogadro/rendering/groupnode.h>
+
+#include <QtCore/QSettings>
+#include <QtWidgets/QDoubleSpinBox>
+#include <QtWidgets/QFormLayout>
+#include <QtWidgets/QVBoxLayout>
+#include <QtWidgets/QWidget>
+
+#include <iostream>
+
+namespace Avogadro {
+namespace QtPlugins {
+
+using Core::Array;
+using Core::Bond;
+using Core::NeighborPerceiver;
+using QtGui::Molecule;
+using QtGui::PluginLayerManager;
+using Rendering::DashedLineGeometry;
+using Rendering::GeometryNode;
+using Rendering::GroupNode;
+
+AtomPairBonds::AtomPairBonds(QObject *p) : ScenePlugin(p)
+{
+  m_layerManager = PluginLayerManager(m_name);
+  
+  QSettings settings;
+  m_angleToleranceDegrees = settings.value("atomPairBonds/angleTolerance", 20.0).toDouble();
```

Comment:
Don't you want both an angle and distance threshold?

Reply:
It also seems pretty easy to add the color and line width as settings here, even if the widget doesn't implement the UI for them yet.

Reply:
Angle and distance, huh... Well, it could be more useful than an angle alone, especially if the distances don't change much across bond types.